RubeesEat/Model/IBillRepository.cs

```diff
@@ -6,4 +6,6 @@ public interface IBillRepository
     void Add(Bill bill);
     IReadOnlyList<Bill> GetAllForUser(Person user);
     IReadOnlyList<BalanceChange> GetRecentBalanceChanges(Person user, int amount);
+    Bill? GetById(Guid guid);
+    bool ContainsKey(Guid guid);
```

Comment:
Given the typing of `GetById` I'd expect it to return `null` if there is no `Bill`. As that is not the case you need a `ContainsKey` to check if there is a certain bill. 

But that leaves the problem that the `Bill` might be deleted/created between `ContainsKey` and `GetById`. I think it would be better to change `GetById` to return `null` when there is no bill found. Then we don't need `ContainsKey` and no longer have a concurrency issue.